middleware.js

```diff
@@ -1,4 +1,38 @@
 var http = require('http');
 
 exports.createServer = function () {
-};
+  var middlewares = [];
+
+  function next(req, res, index) {
+    if(index == undefined) {
+      index = -1;
+    }
+    return function () {
+      for(var i = index + 1; i < middlewares.length; i++) {
+        if(middlewares[i].path == '*' || middlewares[i].path == req.url) {
+          middlewares[i].action.call(this, req, res, next(req, res, i));
```

Comment:
A plain, old style funcall `middlewares[i].action(req, res, next(req, res, i))` should do a good job here.